src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs

```diff
@@ -218,5 +219,36 @@ public void DuplicateRuntimeIdentifiers()
                 .Pass();
 
         }
+
+        [Fact]
+        public void It_Publishes_Successfully_With_RID_Requiring_Properties_And_RuntimeIdentifierS_but_no_RuntimeIdentifier()
+        {
+            var targetFramework = ToolsetInfo.CurrentTargetFramework;
+            var runtimeIdentifier = EnvironmentInfo.GetCompatibleRid(targetFramework);
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld")
+                .WithSource()
+                .WithProjectChanges(project =>
+                {
+                    var ns = project.Root.Name.Namespace;
+                    var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+                    propertyGroup.Add(new XElement(ns + "RuntimeIdentifiers", runtimeIdentifier));
+                    propertyGroup.Add(new XElement(ns + "PublishReadyToRun", "true"));
+                });
```

Comment:
Any reason to start with "HelloWorld" instead of something like this?

```C#
var testProject = new TestProject()
{
    OutputType = "Exe",
    TargetFrameworks = ToolsetInfo.CurrentTargetFramework
};

testProject.AdditionalProperties["RuntimeIdentifiers"] = runtimeIdentifiers;
testProject.AdditionalProperties["PublishReadyToRun"] = "true";
var testAsset = _testAssetsManager.CreateTestProject(testProject);
```

Reply:
Nah, I thought it was the simplest way to make a test project but that's not true.